### PR TITLE
Feature/emmanuelclflores/1036702205 update dist section

### PR DESCRIPTION
**Pull request by wllmwu**
### Administrative Info

Monday Board ID: 1036702205

### Changes

What changes did you make?

- Implement new Distribution section (days of the week with start times, start date, frequency, calendar widget), and new Retail Rescue section (single checkbox), on the agency profile form
    - Add DistributionDays component to encapsulate day-of-the-week checkboxes + time inputs
    - Add TimeBox component for the time inputs
    - Add Calendar component
    - Refactor checkboxes and number boxes to use a direct callback to the form, instead of bouncing between encapsulating components
- Update Agency schema for both sections
- Update validators for agency form data

### Testing

How did you confirm your changes worked?

- Added new agency to the database successfully
- Checked that various bad inputs were rejected
- Tests are ok

### Confirmation of Change

Upload a screenshot, if possible. Otherwise, please provide instructions on how to see the change.

Screen Shot 2021-03-09 at 23 48 19

Screen Shot 2021-03-09 at 21 30 34

### CODE REVIEW

File: frontend/src/pages/AgencyProfileForm/DistributionCalendar/Calendar.jsx [hunk 1]
```diff
@@ -0,0 +1,373 @@
...
+
+    return calendar;
+  };
+
+  /**
+   * Determines if a given date string is a valid distribution date.
+   *
+   * @param {String} date String in default date format to be assessed
+   * @returns Boolean representing if the given date string is a valid
+   * distributiond ate
```
Comment (sblumin) [inline frontend/src/pages/AgencyProfileForm/DistributionCalendar/Calendar.jsx:72]: small typo here! distributiond ate should be distribution date

File: frontend/src/pages/AgencyProfileForm/DistributionCalendar/CalendarStyle.css [hunk 2]
```diff
@@ -0,0 +1,154 @@
+.calendar-container {
```
Comment (sblumin) [inline frontend/src/pages/AgencyProfileForm/DistributionCalendar/CalendarStyle.css:1]: Can you add some inline comments to this file just to break up the different styling and identify larger components

**Review by sblumin — CHANGES_REQUESTED** [review]
Looks great!!!! Incredible job, I only noticed one typo and a couple comments in one css file. Nice job y'all

File: frontend/src/pages/AgencyProfileForm/DistributionDays.jsx [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+              isChecked={item.selected}
+              stateKey={item.stateKey}
+              onChange={onChange}
+            />
+            <TimeBox
+              value={item.time}
+              stateKey={item.timeStateKey}
+              onChange={onChange}
+              valid={validCheck(`distributionStartTimes.${item.stateKey}`)}
+            />
```
Comment (NirmalAgnihotri) [inline frontend/src/pages/AgencyProfileForm/DistributionDays.jsx:42]: Very clean formatting and great use of components.

File: frontend/src/pages/AgencyProfileForm/DistributionCalendar/CalendarStyle.css (hunk 2 above)
Comment (NirmalAgnihotri) [reply frontend/src/pages/AgencyProfileForm/DistributionCalendar/CalendarStyle.css:1]: Seconding this, maybe just add some comments indicating purpose of the styling

**Review by NirmalAgnihotri — APPROVED** [review]
Overall, looks really good and well done. Just add some comments to that CSS file.

**Review by sblumin — APPROVED** [review]
LGTM!

**Pull request merged**